build-scripts/publish_to_maven_local_if_modified.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+# Purpose: Publish android packages to local maven repo, but only if changed since last publish.
+# Dependencies: None
+# Usage: ./automation/publish_to_maven_local_if_modified.py
+
+import os
+import sys
+import time
+import hashlib
+import argparse
+from shared import run_cmd_checked, find_glean_root, fatal_err
+import re
+
+LAST_CONTENTS_HASH_FILE = ".lastAutoPublishContentsHash"
+
+GITIGNORED_FILES_THAT_AFFECT_THE_BUILD = ["local.properties"]
+
+parser = argparse.ArgumentParser(description="Publish android packages to local maven repo, but only if changed since last publish")
+parser.parse_args()
+
+root_dir = find_glean_root()
+if str(root_dir) != os.path.abspath(os.curdir):
+    fatal_err(f"This only works if run from the repo root ({root_dir!r} != {os.path.abspath(os.curdir)!r})")
+
+# This doesn't work on "native" windows, so let's get that out of the way now.
+if sys.platform.startswith("win"):
+    print("NOTE: The autoPublish workflows do not work on native windows.")
```

Comment:
```suggestion
    print("NOTE: The autoPublish workflows do not work on native Windows.")
```

Reply:
👍 (actually I'm not even sure it breaks on Windows, this is copied from a-s, but I guess we can "fix" that in a followup)